source/common/ur_util.h

```diff
@@ -76,4 +80,54 @@ inline bool getenv_tobool( const char* name )
     return 0 == strcmp( "1", env );
 }
 
+inline int parse_env_var(const char *env_var_name,
+                         std::unordered_map<std::string, std::vector<std::string>> &map) {
+    // example: ENV_VAR="param_1:value_1,value_2;param_2:value_1"
+    char main_delim = ';';
+    char key_value_delim = ':';
+    char values_delim = ',';
+    int delim_pos = 0;
+
+    const char *env_var_c_str = std::getenv(env_var_name);
```

Comment:
I think you might want to use the same logic as in `getenv_tobool` to support windows and linux (perhaps we can extract that as a standalone function).